src/test/java/hudson/plugins/view/dashboard/test/MavenConfiguration.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * The MIT License (MIT)
+ *
+ * <p>Copyright (c) 2015
+ *
+ * <p>Permission is hereby granted, free of charge, to any person obtaining a copy of this software
+ * and associated documentation files (the "Software"), to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify, merge, publish, distribute,
+ * sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * <p>The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ *
+ * <p>THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+ * BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package hudson.plugins.view.dashboard.test;
+
+import hudson.tasks.Maven;
+import hudson.tools.InstallSourceProperty;
+import hudson.tools.ZipExtractionInstaller;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.UUID;
+import jenkins.model.Jenkins;
+import org.apache.commons.lang3.ArrayUtils;
+
+public class MavenConfiguration {
+
+  private final Jenkins jenkins;
+
+  public MavenConfiguration(Jenkins jenkins) {
+    this.jenkins = jenkins;
+  }
+
+  public String addInstallation() throws IOException {
+    ZipExtractionInstaller installer =
+        new ZipExtractionInstaller(
+            null,
+            "https://archive.apache.org/dist/maven/maven-3/3.8.4/binaries/apache-maven-3.8.4-bin.tar.gz",
+            "apache-maven-3.8.4");
```

Comment:
More efficient to download the archive from Maven when preparing test resources, as `jenkins-test-harness-tools` does.